storage/src/main/java/org/apache/kafka/storage/internals/log/LogValidator.java

```diff
@@ -574,7 +574,7 @@ private static Optional<ApiRecordError> validateRecord(RecordBatch batch,
         if (batch.magic() <= RecordBatch.MAGIC_VALUE_V1 && batch.isCompressed()) {
             try {
                 record.ensureValid();
-            } catch (InvalidRecordException e) {
+            } catch (InvalidRecordException | CorruptRecordException e) {
```

Comment:
why we need this check?

Reply:
[here](https://github.com/apache/kafka/blob/80f31224aad543dbfc892bce1ad73b6bb693855a/clients/src/main/java/org/apache/kafka/common/record/LegacyRecord.java#L129) is the exception should be thrown so we need to add it.

Reply:
My point was why we need to change the production code in this test migration?

Reply:
I can't enter `recordInvalidChecksums` if I don't have this modification.

Reply:
ok, it seems that is another issue.

#7150 changes the thrown exception from `InvalidRecordException` to `CorruptRecordException`, and so `LogValidator` can't catch it to update `recordInvalidChecksums`. Let me file a Jira for it

Reply:
@TaiJuWu I have assigned https://issues.apache.org/jira/browse/KAFKA-17104 to you

Reply:
Got it, thanks.